File: LeetCode Questions + Tutorials/EngineeringDigest/Leetcode1004.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
function longestOnes(nums, flips) {
    var left = 0, ans = 0, count = 0;
    for (var right = 0; right < nums.length; right++) {
        if (nums[right] == 0) {
            count++;
        }
        while (count > flips) {
            if (nums[left] == 0) {
                count--;
            }
            left++;
        }
        ans = Math.max(ans, right - left + 1);
    }
    return ans;
}
var arr = [0, 0, 1, 1, 0, 0, 1, 1, 1, 0, 1, 1, 0, 0, 0, 1, 1, 1, 1];
var flips = 3;
console.log(longestOnes(arr, flips));

File: LeetCode Questions + Tutorials/EngineeringDigest/Leetcode1004.java
import java.util.Scanner;

public class Leetcode1004 {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the size of the array: ");
        int size = input.nextInt();
        int[] arr = new int[size];
        for (int i = 0; i < size; i++) {
            System.out.printf("Enter %d number element of the array: ", i + 1);
            arr[i] = input.nextInt();
        }
        System.out.print("Enter the value of flips: ");
        int k = input.nextInt();
        input.close();
        System.out.println(longestOnes(arr, k));
    }

    private static int longestOnes(int[] nums, int k) {
        int left = 0, ans = 0, countZero = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == 0) {
                countZero++;
            }
            while (countZero > k) {
                if (nums[left] == 0) {
                    countZero--;
                }
                left++;
            }
            ans = Math.max(ans, i - left + 1);
        }
        return ans;
    }
}